### ISSUE TITLE
Proposal: make material `Stepper` pass proper `isActive` to its children `Step`s, based on `currentStep`

### ISSUE THREAD

**Issue by heejongahn** [flutter/flutter#40554]
## Use case
For now, the users of `Stepper` have to pass duplicate information twice, that is:
1. `currentStep` to `Stepper`
2. `isActive: ([corresponding step of the Step widget] == currentState)` to each `Step`

However, I think it's safe to assume that in major cases, users would want to show `Step` at index `currentStep` to appear as selected one (I mean style-wise). If we implement exactly that, the users of `Stepper` aren't required to pass (redundant, in my opinion) `isActive` to `Step` widget.

## Proposal
Make `Stepper` pass proper `isActive` parameter for children, based on its own `currentStep`. If needed, pass opt-out option for `Step` which disables enabled styling. (I think I can start [here](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/stepper.dart#L512-L533)?)

I'm filing an issue first instead of PR, as I'm not sure if the core team agrees with this change, and/or if there's any possible unintended outcome that I couln't think of. If there's any, please let me know! Thanks for the awesome product.

**Comment by cpboyd** [comment]
I agree that some sort of improved state management for the `Stepper` and individual `Step`s would be nice.

What I'm using currently:
```dart


  StepState _getState(int step) {
    if (_stepErrors[step]) {
      return StepState.error;
    }
    if (_currentStep == step) {
      return StepState.editing;
    }
    if (_currentStep < step) {
      return StepState.indexed;
    }
    return StepState.complete;
  }
```

I've also discovered that since the `Stepper` only renders the current `Step`, any `Form` validation must also be saved hence the `List<bool> _stepErrors`.
